docs/customizing.md

```diff
@@ -245,3 +245,275 @@ Example `pulp_install_plugins` (with bogus version values):
       pulp-rpm:
          version: "7.8.9"
 ```
+
+Deployment Scenarios
```

Comment:
I was originally planning on this being a separate page.

However, customizing.md currently has 2 kind-of-loosely related sections anyway. This is less closely related, but still.

How about we put a Table of Contents at the top? It would only need to display top-level-headers.

Reply:
At the very top? 
I added links to the deployment scenario options. 
I still want to do more work on the installer docs, so will tidy this all up in another PR that won't require your direct input :)